lua/nvim-tree.lua

```diff
@@ -589,17 +590,18 @@ function M.setup(conf)
     return
   end
 
-  if M.setup_called then
-    utils.warn "nvim-tree.lua setup called multiple times"
-    return
-  end
-  M.setup_called = true
-
   legacy.migrate_legacy_options(conf or {})
 
   validate_options(conf)
 
   local opts = merge_options(conf)
+
+  if not opts.suppress_setup_recall_warning and M.setup_called then
+    utils.warn "nvim-tree.lua setup called multiple times"
+    return
```

Comment:
We should return even when `suppress_setup_recall_warning` is set.

nvim-tree will behave unpredictably (duplicate autocommands etc) when setup is run twice.